### ISSUE TITLE
Build only last commit for Gitlab

### ISSUE THREAD

**Issue by kromonos** [php-censor/php-censor#259]
Per default I work with 3 or more branches. master, stage, development and feature/*. Yesterday I merged my development with 13 commits into stage which filled the list of build request quiet fast 😆 
And because it's not enough, it only added the first 6 of 13, so, that PHP-Censor didn't build the latest.
So, it would be really nice to have an option to say "only build latest commit".

**Comment by corpsee** [comment]
@kromonos It is not clear. Could you gave more details please:

* What type of source you use (Git, Github, GItlab etc.)?
* How you start builds (manual, by webhook)?

P. S.: Webhook for Github for example already builds only last commit.

**Comment by kromonos** [comment]
Oh, sorry, my fault. I'm running an own Gitlab install and invoke builds via webhooks.

**Comment by corpsee** [comment]
@kromonos Thanks! Feature request accepted.

**Comment by corpsee** [comment]
@kromonos Could you attach example of Gitlab webhook payload (webhook request body) for push (pull request, tag etc.) without your secure data? Like [here](https://github.com/php-censor/php-censor/issues/13#issuecomment-277923071) for Gogs.

**Comment by kromonos** [comment]
Sure. The push event triggers following:
```
{
  "object_kind": "push",
  "before": "95790bf891e76fee5e1747ab589903a6a1f80f22",
  "after": "da1560886d4f094c3e6c9ef40349f7d38b5d27d7",
  "ref": "refs/heads/master",
  "checkout_sha": "da1560886d4f094c3e6c9ef40349f7d38b5d27d7",
  "user_id": 4,
  "user_name": "John Smith",
  "user_username": "jsmith",
  "user_email": "john@example.com",
  "user_avatar": "https://s.gravatar.com/avatar/d4c74594d841139328695756648b6bd6?s=8://s.gravatar.com/avatar/d4c74594d841139328695756648b6bd6?s=80",
  "project_id": 15,
  "project":{
    "id": 15,
    "name":"Diaspora",
    "description":"",
    "web_url":"http://example.com/mike/diaspora",
    "avatar_url":null,
    "git_ssh_url":"git@example.com:mike/diaspora.git",
    "git_http_url":"http://example.com/mike/diaspora.git",
    "namespace":"Mike",
    "visibility_level":0,
    "path_with_namespace":"mike/diaspora",
    "default_branch":"master",
    "homepage":"http://example.com/mike/diaspora",
    "url":"git@example.com:mike/diaspora.git",
    "ssh_url":"git@example.com:mike/diaspora.git",
    "http_url":"http://example.com/mike/diaspora.git"
  },
  "repository":{
    "name": "Diaspora",
    "url": "git@example.com:mike/diaspora.git",
    "description": "",
    "homepage": "http://example.com/mike/diaspora",
    "git_http_url":"http://example.com/mike/diaspora.git",
    "git_ssh_url":"git@example.com:mike/diaspora.git",
    "visibility_level":0
  },
  "commits": [
    {
      "id": "b6568db1bc1dcd7f8b4d5a946b0b91f9dacd7327",
      "message": "Update Catalan translation to e38cb41.",
      "timestamp": "2011-12-12T14:27:31+02:00",
      "url": "http://example.com/mike/diaspora/commit/b6568db1bc1dcd7f8b4d5a946b0b91f9dacd7327",
      "author": {
        "name": "Jordi Mallach",
        "email": "jordi@softcatala.org"
      },
      "added": ["CHANGELOG"],
      "modified": ["app/controller/application.rb"],
      "removed": []
    },
    {
      "id": "da1560886d4f094c3e6c9ef40349f7d38b5d27d7",
      "message": "fixed readme",
      "timestamp": "2012-01-03T23:36:29+02:00",
      "url": "http://example.com/mike/diaspora/commit/da1560886d4f094c3e6c9ef40349f7d38b5d27d7",
      "author": {
        "name": "GitLab dev user",
        "email": "gitlabdev@dv6700.(none)"
      },
      "added": ["CHANGELOG"],
      "modified": ["app/controller/application.rb"],
      "removed": []
    }
  ],
  "total_commits_count": 4
}
```
You can find examples for other events at https://docs.gitlab.com/ee/user/project/integrations/webhooks.html

**Comment by corpsee** [comment]
@kromonos Thanks!

**Comment by corpsee** [comment]
@kromonos Done. Gitlab now builds only last commit like Github (In the branch: `1.1-gitlab-build-only-last-commit`). Could you test the branch `1.1-gitlab-build-only-last-commit` on your Gitlab installation please?

**Comment by kromonos** [comment]
Do I need to setup something or is it active per default?

**Comment by corpsee** [comment]
@kromonos It works by default.

**Comment by kromonos** [comment]
Works perfectly :)

**Comment by corpsee** [comment]
@kromonos Thank you for the testing! Changes merged into `master` branch and will be released with version `1.1.0`.
